### PR TITLE
Ensure that IAM Users have tags

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: exports.js [hunk 1]
```diff
@@ -298,6 +298,7 @@ module.exports = {
         'iamUserAdmins'                 : require(__dirname + '/plugins/aws/iam/iamUserAdmins.js'),
         'iamUserNameRegex'              : require(__dirname + '/plugins/aws/iam/iamUserNameRegex.js'),
         'iamUserUnauthorizedToEdit'     : require(__dirname + '/plugins/aws/iam/iamUserUnauthorizedToEdit'),
+        'iamUsersHasTags'               : require(__dirname + '/plugins/aws/iam/iamUsersHasTags.js'),
```
Comment (AkhtarAmir) [inline exports.js:301]:
```suggestion
        'iamUserHasTags'                : require(__dirname + '/plugins/aws/iam/iamUserHasTags.js'),
```

Comment (AkhtarAmir) [inline exports.js:301]:
```suggestion
        'iamUserHasTags'                : require(__dirname + '/plugins/aws/iam/iamUserHasTags.js'),
```

File: plugins/aws/iam/iamUsersHasTags.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+
+var helpers = require('../../../helpers/aws');
+      
+module.exports = {
+    title: 'IAM Users Has Tags',
```
Comment (AkhtarAmir) [inline plugins/aws/iam/iamUsersHasTags.js:5]:
```suggestion
    title: 'IAM User Has Tags',
```

File: plugins/aws/iam/iamUsersHasTags.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+
+var helpers = require('../../../helpers/aws');
+
```
Comment (AkhtarAmir) [inline plugins/aws/iam/iamUsersHasTags.js:3]: Also need to change file name

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: exports.js [hunk 4]
```diff
@@ -298,6 +298,7 @@ module.exports = {
         'iamUserAdmins'                 : require(__dirname + '/plugins/aws/iam/iamUserAdmins.js'),
         'iamUserNameRegex'              : require(__dirname + '/plugins/aws/iam/iamUserNameRegex.js'),
         'iamUserUnauthorizedToEdit'     : require(__dirname + '/plugins/aws/iam/iamUserUnauthorizedToEdit'),
+        'iamUserHasTags'               : require(__dirname + '/plugins/aws/iam/iamUserHasTags.js'),
```
Comment (alphadev4) [inline exports.js:301]:
```suggestion
        'iamUserHasTags'                : require(__dirname + '/plugins/aws/iam/iamUserHasTags.js'),
```

File: plugins/aws/iam/iamUserHasTags.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    description:  'Ensure that AWS IAM Users have tags associated.',
+    more_info: 'Tags help you to group resources together that are related to or associated with each other. It is a best practice to tag cloud resources to better organize and gain visibility into their usage.',
+    link: 'https://docs.aws.amazon.com/IAM/latest/UserGuide/id_tags_users.html',
+    recommended_action: 'Modify IAM User and add tags',
+    apis: ['IAM:listUsers'],
+          
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+
```
Comment (alphadev4) [inline plugins/aws/iam/iamUserHasTags.js:17]:
```suggestion
```

File: plugins/aws/iam/iamUserHasTags.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        }
+    
+        if (!listUsers.data.length) {
+            helpers.addResult(results, 0, 'No iam users found', 'global');
+        } 
+        for (var user of listUsers.data) {
+            if (!user.Tags.length){
+                helpers.addResult(results, 2, 'IAM User does not have Tags', 'global', user.Arn);
+            } else {
+                helpers.addResult(results, 0, 'IAM User have Tags specified', 'global', user.Arn);
```
Comment (alphadev4) [inline plugins/aws/iam/iamUserHasTags.js:37]:
```suggestion
                helpers.addResult(results, 0, 'IAM User has tags', 'global', user.Arn);
```

File: plugins/aws/iam/iamUserHasTags.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    
+        if (!listUsers.data.length) {
+            helpers.addResult(results, 0, 'No iam users found', 'global');
+        } 
+        for (var user of listUsers.data) {
+            if (!user.Tags.length){
+                helpers.addResult(results, 2, 'IAM User does not have Tags', 'global', user.Arn);
+            } else {
+                helpers.addResult(results, 0, 'IAM User has tags', 'global', user.Arn);
+
```
Comment (alphadev4) [inline plugins/aws/iam/iamUserHasTags.js:38]:
```suggestion
```

File: plugins/aws/iam/iamUserHasTags.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        if (listUsers.err || !listUsers.data) {
+            helpers.addResult(results, 3,
+                'Unable to query for iam users: ' + helpers.addError(listUsers));
+            return callback(null, results, source);
+        }
+    
+        if (!listUsers.data.length) {
+            helpers.addResult(results, 0, 'No iam users found', 'global');
+        } 
+        for (var user of listUsers.data) {
```
Comment (alphadev4) [inline plugins/aws/iam/iamUserHasTags.js:33]:
```suggestion

        for (var user of listUsers.data) {
```

**Review by alphadev4 — APPROVED** [review]

**Review by AkhtarAmir — APPROVED** [review]

**Pull request merged**